internal/querynode/impl.go

```diff
@@ -485,6 +485,21 @@ func (node *QueryNode) LoadSegments(ctx context.Context, in *querypb.LoadSegment
 		return node.TransferLoad(ctx, in)
 	}
 
+	alreadyLoaded := true
+	for _, info := range in.Infos {
+		has, _ := node.metaReplica.hasSegment(info.SegmentID, segmentTypeSealed)
+		if !has {
+			alreadyLoaded = false
+		}
+	}
```

Comment:
cannot skip all segment info here, 2.2 has task merge logic

Reply:
only all segment has been loaded , it can skip